api_audio_darwin.go

```diff
@@ -0,0 +1,103 @@
+package oto
+
+import _ "runtime/cgo" // TODO: remove once purego(#12) is solved.
+import (
+	"unsafe"
+
+	"github.com/ebitengine/purego"
+)
+
+const (
+	avAudioSessionErrorCodeCannotStartPlaying = 0x21706c61 // '!pla'
+	avAudioSessionErrorCodeSiriIsRecording    = 0x73697269 // 'siri'
+)
+
+const (
+	kAudioFormatLinearPCM = 0x6C70636D //'lpcm'
+)
+
+const (
+	kAudioFormatFlagIsFloat = 1 << 0 // 0x1
+)
+
+type _AudioStreamBasicDescription struct {
+	mSampleRate       float64
+	mFormatID         uint32
+	mFormatFlags      uint32
+	mBytesPerPacket   uint32
+	mFramesPerPacket  uint32
+	mBytesPerFrame    uint32
+	mChannelsPerFrame uint32
+	mBitsPerChannel   uint32
+	mReserved         uint32
+}
+
+type _AudioQueueRef uintptr
+
+type _AudioTimeStamp uintptr
+
+type _AudioStreamPacketDescription struct {
+	mStartOffset            int64
+	mVariableFramesInPacket uint32
+	mDataByteSize           uint32
+}
+
+type _AudioQueueBufferRef *_AudioQueueBuffer
+
+type _AudioQueueBuffer struct {
+	mAudioDataBytesCapacity uint32
+	mAudioData              uintptr // void*
+	mAudioDataByteSize      uint32
+	mUserData               uintptr // void*
+
+	mPacketDescriptionCapacity uint32
+	mPacketDescriptions        *_AudioStreamPacketDescription
+	mPacketDescriptionCount    uint32
+}
+
+type _AudioQueueOutputCallback func(inUserData unsafe.Pointer, inAQ _AudioQueueRef, inBuffer _AudioQueueBufferRef)
+
+var (
+	toolbox                    = purego.Dlopen("/System/Library/Frameworks/AudioToolbox.framework/Versions/Current/AudioToolbox", purego.RTLD_GLOBAL)
+	atAudioQueueNewOutput      = purego.Dlsym(toolbox, "AudioQueueNewOutput")
+	atAudioQueueStart          = purego.Dlsym(toolbox, "AudioQueueStart")
+	atAudioQueuePause          = purego.Dlsym(toolbox, "AudioQueuePause")
+	atAudioQueueAllocateBuffer = purego.Dlsym(toolbox, "AudioQueueAllocateBuffer")
+	atAudioQueueEnqueueBuffer  = purego.Dlsym(toolbox, "AudioQueueEnqueueBuffer")
+)
+
+func _AudioQueueNewOutput(inFormat *_AudioStreamBasicDescription, inCallbackProc _AudioQueueOutputCallback, inUserData unsafe.Pointer, inCallbackRunLoop uintptr, inCallbackRunLoopMod uintptr, inFlags uint32, outAQ *_AudioQueueRef) uintptr {
+	ret, _, _ := purego.SyscallN(atAudioQueueNewOutput,
```

Comment:
So we can ignore errors from SyscallN, right?

Reply:
We can since frameworks don't return errors in the standard way